python/paddle/fluid/tests/unittests/mkldnn/test_shape_mkldnn_op.py

```diff
@@ -0,0 +1,85 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+from paddle.fluid.tests.unittests.op_test import OpTest, OpTestTool
+import paddle
+from paddle.fluid import core
+from paddle.fluid.op import Operator
+
+
+@OpTestTool.skip_if_not_cpu_bf16()
+class TestShapeOneDNNOp(OpTest):
+    def setUp(self):
+        self.op_type = "shape"
+        self.config()
+        self.attrs = {'use_mkldnn': True}
+        self.inputs = {'Input': np.zeros(self.shape).astype(self.dtype)}
+        self.outputs = {'Out': np.array(self.shape)}
+
+    def config(self):
+        self.shape = [5, 7, 4]
+        self.dtype = np.float32
+
+    def test_check_output(self):
+        self.check_output_with_place(core.CPUPlace())
+
+
+class TestShape1DOneDNNOp(TestShapeOneDNNOp):
```

Comment:
Every time I see such almost identical test classes I wonder why python `unittest` package does not support parameterization. This is rather just a note for you to be aware of those things. The best solution for such cases would be to use `pytest` package since it supports [parameterized test cases](https://docs.pytest.org/en/latest/parametrize.html). However uncle Google says that there is also package like [ddt](https://stackoverflow.com/a/50514671) which looks promising and can be used for `unittest` package.